frontend/src/components/AllQueues.js

```diff
@@ -80,35 +88,86 @@ function QueueTab(props) {
   )
 }
 
-function AllQueues() {
+const AllQueues = observer(() => {
+  const queueStore = useContext(QueueStoreContext)
+  const queueSize = Object.keys(queueStore.queueIds).length
+
+  useEffect(() => {
+    for (let i = 1; i <= queueSize; i++) queueStore.updateQueue(i)
+  })
   const classes = useStyles()
-  const [value, setValue] = React.useState(0)
+  const [value, setValue] = useState(0)
   function handleChange(event, newValue) {
     setValue(newValue)
   }
 
   return (
     <div className={classes.root}>
-      <Tabs variant="fullWidth" value={value} onChange={handleChange}>
-        <QueueTab
-          className={classes.queueTab}
-          queueData={caseManagementQueueData}
-        />
-        <QueueTab
-          className={classes.queueTab}
-          queueData={legalServicesQueueData}
-        />
-        <QueueTab className={classes.queueTab} queueData={stepQueueData} />
-      </Tabs>
-      {value === 0 && <QueueTable queueData={caseManagementQueueData} />}
-      {value === 1 && <QueueTable queueData={legalServicesQueueData} />}
-      {value === 2 && <QueueTable queueData={stepQueueData} />}
+      <AppBar position="static" color="default">
+        <Tabs
+          value={value}
+          onChange={handleChange}
+          variant="scrollable"
+          scrollButtons="on"
+          indicatorColor="primary"
+          textColor="primary"
+          aria-label="scrollable force tabs example"
+        >
+          <QueueTab
+            className={classes.queueTab}
+            queueData={queueStore.queueIds["TESTING"]}
+          />
+          <QueueTab
+            className={classes.queueTab}
+            queueData={queueStore.queueIds["CM"]}
+          />
+          <QueueTab
+            className={classes.queueTab}
+            queueData={queueStore.queueIds["SSHP"]}
+          />
+          <QueueTab
+            className={classes.queueTab}
+            queueData={queueStore.queueIds["LEGAL"]}
+          />
+          <QueueTab
+            className={classes.queueTab}
+            queueData={queueStore.queueIds["CRAFT"]}
+          />
+          <QueueTab
+            className={classes.queueTab}
+            queueData={queueStore.queueIds["PHAN"]}
+          />
+          <QueueTab
+            className={classes.queueTab}
+            queueData={queueStore.queueIds["STEP"]}
+          />
+          <QueueTab
+            className={classes.queueTab}
+            queueData={queueStore.queueIds["BIENSTAR"]}
+          />
+          <QueueTab
+            className={classes.queueTab}
+            queueData={queueStore.queueIds["SKWC"]}
+          />
+        </Tabs>
+      </AppBar>
+      {value === 0 && <QueueTable queueData={queueStore.queueIds["TESTING"]} />}
```

Comment:
Could we map each `value` with a `queueId` instead?

```js
const valueToQueueId = { 0: 'TESTING', 1: 'CM', 2: 'SSHP', ... , 8: 'SKWC' }

...

<QueueTable queueData={queueStore.queueIds[valueToQueueId[value]]} />
```

And perhaps we could rename `value` to `tabValue` to be more descriptive/explicit

Reply:
I renamed "value" "tabValue".  

Regarding reversing the map, the mapping you describe is already in queueStore in queueStats.  We could simplify considerably by setting the tab numbers in store and just setting them up in the component by number. Making the tabs is just a for loop, and we don't need all the if's at the end. 

I did it like this because I thought it would be better to set the order in the component.

Reply:
I didn't much like that queueIds and queueStats have overlapping data, but I thought it made things easier to understand

Reply:
Open to changing it

Reply:
It's a lot better simplified. Changed it.  Getting rid of queueIds in queueStore, i think